File: Minho/0802_2.cpp
#include <iostream>
using namespace std;

int N, M;
int arr[10];
int result[10];

void DFS(int index, int cnt) {
    if (cnt == M) {
        for (int i = 1; i <= M; i++) {
            cout << result[i] << " ";
        }
        cout << '\n';
        return;
    }

    for (int i = index + 1; i <= N; i++) {
        result[cnt + 1] = i;
        DFS(i, cnt + 1);
    }
}

int main() {
    cin >> N >> M;
    for (int i = 1; i <= N; i++) {
        arr[i] = i;
    }

    for (int i = 1; i <= N; i++) {
        result[1] = i;
        DFS(i, 1);
    }
}

File: subin/0802/0802_2.java
package fc0801;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class nm2 {
    public static int N,M;
    public static boolean[] isVisited;
    public static int[] arr;

    public static void main(String[] args) throws IOException {
        BufferedReader br= new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st= new StringTokenizer(br.readLine());
        N= Integer.parseInt(st.nextToken());
        M= Integer.parseInt(st.nextToken());

        isVisited=new boolean[N];
        arr=new int[M];

        dfs(0,0);
    }

    public static void dfs(int cur,int cnt){
        if(cnt==M){
            for (int i = 0; i < M; i++) {
                System.out.print(arr[i] + " ");
            }
            System.out.println();
            return;
        }

        for(int i=cur;i<N;i++){
            if(!isVisited[i]){
                isVisited[i]=true;
                arr[cnt]=i+1;
                dfs(i+1,cnt+1);
                isVisited[i]=false;
            }
        }
    }
}
